app/com/linkedin/drelephant/ElephantRunner.java

```diff
@@ -73,6 +80,17 @@ private void loadGeneralConfiguration() {
     _executorNum = Utils.getNonNegativeInt(configuration, EXECUTOR_NUM_KEY, EXECUTOR_NUM);
     _fetchInterval = Utils.getNonNegativeLong(configuration, FETCH_INTERVAL_KEY, FETCH_INTERVAL);
     _retryInterval = Utils.getNonNegativeLong(configuration, RETRY_INTERVAL_KEY, RETRY_INTERVAL);
+
+    if (Utils.isSet(RETENTION_PERIOD_KEY)) {
+      _retentionPeriod = configuration.getInt(RETENTION_PERIOD_KEY, 0);
```

Comment:
use Utils.getNonNegativeInt instead?

Reply:
i don't need to reset it if it's negative... there is no "default" for this. but i can remove the check if it's set...